### PR TITLE
Exit addition

### PR DESCRIPTION
**Pull request by drozdziak1**
* [x] Local addition via POST `/exits`
* [x] Hauling an exit list from a remote HTTPS server
* [x] Docs

### CODE REVIEW

File: rita/src/rita_client/dashboard/network_endpoints.rs [hunk 1]
```diff
@@ -370,3 +372,102 @@ pub fn remote_logging_level(path: Path<String>) -> Box<Future<Item = HttpRespons
...
+}
+
+pub fn exits_sync(
+    list_url_json: Json<HashMap<String, String>>,
+) -> Box<Future<Item = HttpResponse, Error = Error>> {
+    debug!("/exits/sync hit with {:?}", list_url_json);
+
+    let list_url = match list_url_json.get("url") {
+        Some(url) if url.starts_with("https://") => url,
+        Some(_unsafe_url) => {
```
Comment (mpapierski) [inline rita/src/rita_client/dashboard/network_endpoints.rs]: Wouldn't it make sense to match "http://", and another match that would say that the exit url is invalid?

```suggestion
        Some(url) if url.starts_with("http://") => {
```

File: rita/src/rita_client/dashboard/network_endpoints.rs [hunk 2]
```diff
@@ -370,3 +372,102 @@ pub fn remote_logging_level(path: Path<String>) -> Box<Future<Item = HttpRespons
...
+
+                return Box::new(future::ok(
+                    HttpResponse::new(StatusCode::BAD_REQUEST)
+                        .into_builder()
+                        .json(ret),
+                ));
+            }
+        },
+        Err(e) => {
+            let mut ret = HashMap::<String, String>::new();
```
Comment (mpapierski) [inline rita/src/rita_client/dashboard/network_endpoints.rs]: I think in this case rust can *probably* will infer type of HashMap, and other similiar cases.

```suggestion
            let mut ret = HashMap::new();
```

File: docs/api/router-dashboard.md [hunk 3]
```diff
@@ -106,6 +175,45 @@ This file documents the dashboard API found in Rita client.
...
+  - `500 Internal Server Error` when the request fails
+```json
+{
+  "error": "<description>",
+  "rust_error": "<stringified_rust_error>"
+}```
+
+- Sample Call:
+
+`curl 127.0.0.1:4877/exits/borked/reset -H "Content-Type:
```
Comment (mpapierski) [inline docs/api/router-dashboard.md]: Accidentally wrapped in the middle of command?

Comment (drozdziak1) [reply docs/api/router-dashboard.md]: yep

File: rita/src/rita_client/dashboard/network_endpoints.rs (hunk 2 above)
Comment (drozdziak1) [reply rita/src/rita_client/dashboard/network_endpoints.rs]: That's true, needed this for a dummy version, but that's not the case anymore

**Pull request merged**